src/utils.rs

```diff
@@ -210,6 +266,44 @@ mod tests {
             load_wallet(&wallet_path).unwrap();
         })
     }
+
+    #[test]
+    // case: wallet path not exist
+    fn test_ensure_no_wallet_exists_no_wallet() {
+        with_tmp_dir(|tmp_dir| {
+            let wallet_path = tmp_dir.join("wallet.json");
+            remove_wallet(&wallet_path);
+            ensure_no_wallet_exists(&wallet_path, false, &INPUT_NOP[..]).unwrap();
+        });
+    }
+
+    #[test]
+    #[should_panic]
+    // case: wallet path exist without --force and input[no]
+    fn test_ensure_no_wallet_exists_throws_err() {
+        with_tmp_dir(|tmp_dir| {
+            let wallet_path = tmp_dir.join("wallet.json");
+            create_wallet(&wallet_path);
+            ensure_no_wallet_exists(&wallet_path, false, &INPUT_NO[..]).unwrap();
+        });
+    }
+
+    #[test]
+    // case: wallet path exist
+    fn test_ensure_no_wallet_exists_exists_wallet() {
+        // case: wallet path exist without --force and input[yes]
+        with_tmp_dir(|tmp_dir| {
+            let wallet_path = tmp_dir.join("wallet.json");
+            create_wallet(&wallet_path);
+            ensure_no_wallet_exists(&wallet_path, false, &INPUT_YES[..]).unwrap();
+        });
+        // case: wallet path exist with --force
+        with_tmp_dir(|tmp_dir| {
+            let wallet_path = tmp_dir.join("wallet.json");
+            create_wallet(&wallet_path);
+            ensure_no_wallet_exists(&wallet_path, true, &INPUT_NOP[..]).unwrap();
+        });
+    }
```

Comment:
This is great! Let's also add a test case where a wallet exists at `wallet.json`, the user specified a different path like `custom-wallet.json`, and does not supply `force`. In that case, there should be no warning or error.